### PR TITLE
creditbureau configuration

### PR DESCRIPTION
**Pull request by nikpawar89**

### CODE REVIEW

**Comment by rajuan** [comment]
Please resolve the conflicts and send the PR again. We plan to review this next week, please provide us with requirement/design specs to help us with the review

**Comment by nazeer1100126** [comment]
Remove all commented code

File: fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java [hunk 1]
```diff
@@ -0,0 +1,350 @@
+package org.apache.fineract.CreditCheck.api;
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java:1]: package name should be in lower case

File: fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+        this.toApiJsonSerializer=toApiJsonSerializer;
+        this.commandsSourceWritePlatformService=commandsSourceWritePlatformService;
+    }
+    
+    
+    
+    @GET
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    public String getCreditBureau(@Context final UriInfo uriInfo)
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java:108]: This method is returning list of CreditBureauData. Change the method name to reflect it

File: fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+        final ApiRequestJsonSerializationSettings settings = this.apiRequestParameterHelper.process(uriInfo.getQueryParameters());
+        return this.toApiJsonSerializer.serialize(settings, creditBureau, this.RESPONSE_DATA_PARAMETERS);
+       
+    }
+    
+  @GET
+    @Path("/mappings")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    public String getCreditBureauLPMapping(@Context final UriInfo uriInfo)
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java:123]: LP should be LoanProduct

File: fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+       
+   
+           
+    }
+  
+  @GET
+  @Path("/orgcb")
+  @Consumes({ MediaType.APPLICATION_JSON })
+  @Produces({ MediaType.APPLICATION_JSON })
+  public String getOrganisationCreditBureau(@Context final UriInfo uriInfo)
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java:140]: Method name should reflect that it is returning list of objects

File: fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+
+
+@Path("/cbconfig")
+@Component
+@Scope("singleton")
+public class CreditBureauConfigurationAPI
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/CreditCheck/api/CreditBureauConfigurationAPI.java:42]: I suggest to have different API resources to handle org and cbconfig.

File: fineract-provider/src/main/resources/sql/migrations/core_db/V312__Creditbureau_configuration.sql [hunk 6]
```diff
@@ -0,0 +1,61 @@
+
+
+CREATE TABLE `m_creditbureau` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `name` varchar(100) NOT NULL,
+  `product` varchar(100) NOT NULL,
+  `country` varchar(100) NOT NULL,
+  `implementationKey` varchar(100) NOT NULL,
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V312__Creditbureau_configuration.sql:8]: Do we need to configure implementationKey for every product? or it is same for all products under some credit bureau name?

File: fineract-provider/src/main/resources/sql/migrations/core_db/V312__Creditbureau_configuration.sql [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `unique impl` (`name`,`product`,`country`,`implementationKey`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+CREATE TABLE `m_organisation_creditbureau` (
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V312__Creditbureau_configuration.sql:14]: What is the use of this table? Under same tenant can we have more than one organization alias?

File: fineract-provider/src/main/resources/sql/migrations/core_db/V312__Creditbureau_configuration.sql [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+
+
+CREATE TABLE `m_creditbureau_configuration` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `configkey` varchar(50) DEFAULT NULL,
+  `value` varchar(50) DEFAULT NULL,
+  `organisation_creditbureau_id` bigint(20) DEFAULT NULL,
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V312__Creditbureau_configuration.sql:32]: Is this config values vary based on alias table?

File: fineract-provider/src/main/resources/META-INF/spring/appContext.xml [hunk 9]
```diff
@@ -79,6 +81,9 @@
 	<jpa:repositories base-package="org.apache.fineract.mix.domain" />
 	<jpa:repositories base-package="org.apache.fineract.scheduledjobs.domain" />
 	<jpa:repositories base-package="org.apache.fineract.template.domain" />
+	<jpa:repositories base-package="com.conflux.highmark.domain" />
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/META-INF/spring/appContext.xml:84]: remove conflux package names

File: fineract-provider/src/main/resources/META-INF/spring/appContext.xml [hunk 10]
```diff
@@ -54,7 +54,9 @@
 										  org.apache.fineract.template.*,
 										  org.apache.fineract.template.service.*,
 										  org.apache.fineract.useradministration.*,
-										  org.apache.fineract.batch">
+										  org.apache.fineract.batch,
+										  org.apache.fineract.CreditCheck.*,
+										  com.conflux.highmark.*">
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/META-INF/spring/appContext.xml:59]: please remove conflux package names

File: fineract-provider/src/main/java/org/apache/fineract/useradministration/service/RoleReadPlatformService.java [hunk 11]
```diff
@@ -18,10 +18,10 @@
  */
 package org.apache.fineract.useradministration.service;
 
-import org.apache.fineract.useradministration.data.RoleData;
-
 import java.util.Collection;
 
+import org.apache.fineract.useradministration.data.RoleData;
+
 public interface RoleReadPlatformService {
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/useradministration/service/RoleReadPlatformService.java:25]: This file should not be included in PR as there is no change.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/FieldConfigurationReadPlatformServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+import org.apache.fineract.infrastructure.core.service.RoutingDataSource;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.portfolio.clientaddress.data.FieldConfigurationData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FieldConfigurationReadPlatformServiceImpl implements FieldConfigurationReadPlatformService
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/FieldConfigurationReadPlatformServiceImpl.java:17]: is FieldConfigurationReadPlatformServiceImpl belongs this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/FieldConfigurationReadPlatformService.java [hunk 13]
```diff
@@ -0,0 +1,15 @@
+package org.apache.fineract.portfolio.clientaddress.service;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.fineract.portfolio.clientaddress.data.FieldConfigurationData;
+
+public interface FieldConfigurationReadPlatformService
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/FieldConfigurationReadPlatformService.java:8]: is FieldConfigurationReadPlatformService belongs this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/ClientAddressReadPlatformServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+import org.apache.fineract.infrastructure.core.service.RoutingDataSource;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.portfolio.clientaddress.data.ClientAddressData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ClientAddressReadPlatformServiceImpl implements ClientAddressReadPlatformService
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/ClientAddressReadPlatformServiceImpl.java:16]: is ClientAddressReadPlatformServiceImpl beongs to this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/ClientAddressReadPlatformService.java [hunk 15]
```diff
@@ -0,0 +1,11 @@
+package org.apache.fineract.portfolio.clientaddress.service;
+
+import java.util.Collection;
+
+import org.apache.fineract.portfolio.clientaddress.data.ClientAddressData;
+
+public interface ClientAddressReadPlatformService {
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/ClientAddressReadPlatformService.java:7]: is ClientAddressReadPlatformService belongs to this PR

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/AddressWritePlatformServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+import org.apache.fineract.portfolio.clientaddress.domain.ClientAddressRepositoryWrapper;
+import org.apache.fineract.portfolio.clientaddress.serialization.AddAddressCommandFromApiJsonDeserializer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+@Service
+public class AddressWritePlatformServiceImpl implements AddressWritePlatformService
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/AddressWritePlatformServiceImpl.java:26]: is AddressWritePlatformServiceImpl belongs to this PR

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/AddressWritePlatformService.java [hunk 17]
```diff
@@ -0,0 +1,12 @@
+package org.apache.fineract.portfolio.clientaddress.service;
+
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResult;
+import org.apache.fineract.portfolio.client.domain.Client;
+
+public interface AddressWritePlatformService
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/AddressWritePlatformService.java:7]: is AddressWritePlatformService belongs to this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/AddressReadPlatformServiceImpl.java [hunk 18]
```diff
@@ -0,0 +1,248 @@
...
+import org.apache.fineract.infrastructure.core.service.RoutingDataSource;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.portfolio.clientaddress.data.AddressData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AddressReadPlatformServiceImpl implements AddressReadPlatformService
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/AddressReadPlatformServiceImpl.java:22]: is AddressReadPlatformServiceImpl belongs to this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/AddressReadPlatformService.java [hunk 19]
```diff
@@ -0,0 +1,20 @@
+package org.apache.fineract.portfolio.clientaddress.service;
+
+import java.util.Collection;
+
+import org.apache.fineract.portfolio.clientaddress.data.AddressData;
+
+public interface AddressReadPlatformService
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/service/AddressReadPlatformService.java:7]: is AddressReadPlatformService belongs to this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/serialization/AddAddressCommandFromApiJsonDeserializer.java [hunk 20]
```diff
@@ -0,0 +1,269 @@
...
+import org.apache.fineract.portfolio.clientaddress.data.FieldConfigurationData;
+import org.apache.fineract.portfolio.clientaddress.service.FieldConfigurationReadPlatformService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.gson.JsonElement;
+import com.google.gson.reflect.TypeToken;
+
+@Component
+public class AddAddressCommandFromApiJsonDeserializer
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/serialization/AddAddressCommandFromApiJsonDeserializer.java:27]: is AddAddressCommandFromApiJsonDeserializer belongs to this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/handler/UpdateClientAddressCommandHandler.java [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+import org.apache.fineract.commands.handler.NewCommandSourceHandler;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResult;
+import org.apache.fineract.portfolio.clientaddress.service.AddressWritePlatformService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@CommandType(entity="ADDRESS",action="UPDATE")
+public class UpdateClientAddressCommandHandler implements NewCommandSourceHandler
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/handler/UpdateClientAddressCommandHandler.java:13]: is UpdateClientAddressCommandHandler belongs to this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/handler/AddClientAddressCommandHandler.java [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+import org.apache.fineract.commands.annotation.CommandType;
+import org.apache.fineract.commands.handler.NewCommandSourceHandler;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResult;
+import org.apache.fineract.portfolio.clientaddress.service.AddressWritePlatformService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+@Service
+@CommandType(entity="ADDRESS",action="CREATE")
+public class AddClientAddressCommandHandler implements NewCommandSourceHandler
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/handler/AddClientAddressCommandHandler.java:12]: is AddClientAddressCommandHandler belongs to this PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/exception/AddressNotFoundException.java [hunk 23]
```diff
@@ -0,0 +1,18 @@
+package org.apache.fineract.portfolio.clientaddress.exception;
+
+import org.apache.fineract.infrastructure.core.exception.AbstractPlatformResourceNotFoundException;
+
+public class AddressNotFoundException extends AbstractPlatformResourceNotFoundException {
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/exception/AddressNotFoundException.java:5]: is AddressNotFoundException belongs to this PR

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/domain/FieldConfigurationRepository.java [hunk 24]
```diff
@@ -0,0 +1,10 @@
+package org.apache.fineract.portfolio.clientaddress.domain;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+public interface FieldConfigurationRepository extends JpaRepository<FieldConfiguration,Long>,
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/clientaddress/domain/FieldConfigurationRepository.java:6]: is FieldConfigurationRepository belongs to this PR?

**Comment by nazeer1100126** [comment]
I see lot of files doesn't belongs to this PR. Please remove them and send the PR again.

**Comment by nikpawar89** [comment]
Closing this PR as a new PR has been raised for the requirement:
https://github.com/apache/incubator-fineract/pull/320

**Pull request closed without merging**